app/controllers/users_controller.rb

```diff
@@ -0,0 +1,56 @@
+class UsersController < ApplicationController
+  skip_before_action :authenticate, only: [:index, :new, :create]
+
```

Comment:
Are you not allowing a user to view another user's profile page?  Find it curious that you only use the session id to find a user.